sql/moz-fx-data-shared-prod/subscription_platform/nonprod_stripe_subscriptions_history/view.sql

```diff
@@ -0,0 +1,294 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.subscription_platform.nonprod_stripe_subscriptions_history`
+AS
+WITH subscriptions_history AS (
+  SELECT
+    customer_id,
+    id AS subscription_id,
+    _fivetran_synced AS synced_at,
+    _fivetran_start AS valid_from,
+    LEAD(_fivetran_start) OVER (PARTITION BY id ORDER BY _fivetran_start) AS valid_to,
+    created,
+    trial_start,
+    trial_end,
+    COALESCE(trial_end, start_date) AS subscription_start_date,
+    cancel_at,
+    cancel_at_period_end,
+    canceled_at,
+    JSON_VALUE(metadata, "$.cancelled_for_customer_at") AS canceled_for_customer_at,
+    ended_at,
+    status,
+    TIMESTAMP_SECONDS(
+      CAST(JSON_VALUE(metadata, "$.plan_change_date") AS INT64)
+    ) AS plan_change_date,
+    JSON_VALUE(metadata, "$.previous_plan_id") AS previous_plan_id,
+  FROM
+    `dev-fivetran`.stripe_nonprod.subscription_history
+),
+subscriptions_history_with_lead_plan_metadata AS (
+  SELECT
+    *,
+    LEAD(plan_change_date) OVER (
+      PARTITION BY
+        subscription_id
+      ORDER BY
+        valid_from
+    ) AS lead_plan_change_date,
+    LEAD(previous_plan_id) OVER (
+      PARTITION BY
+        subscription_id
+      ORDER BY
+        valid_from
+    ) AS lead_previous_plan_id,
+  FROM
+    subscriptions_history
+),
+subscriptions_history_with_plan_ends AS (
+  SELECT
+    *,
+    -- A new `previous_plan_id` value means the previous row was the last row with that plan.
```

Comment:
This comment matches the behavior for `LAG`, but query logic here and in subsequent CTEs is consitent with `LEAD`, so:
```suggestion
    -- A new `previous_plan_id` value means the current row is the last row with that plan.
```

Reply:
I think the original comment is accurate, though I'll admit the logic of using `LEAD()` with a field named `previous_...` is difficult to explain unambiguously.  The crux is that when a new `previous_plan_id` is set the row has already been updated for the new plan, so we want the previous row to that change happening.

Maybe it'd be clearer to rephrase it from the perspective of the current row?
```suggestion
    -- If the next row has a different `previous_plan_id` then this was the last row with that plan.
```

Reply:
your suggestion is much more clear, thank you